Formal2024/Homeworks/homework2.lean

```diff
@@ -0,0 +1,79 @@
+/-
+# Homework 1: Logic
+References: [MIL] Mathematics in Lean, [Tut] Tutorials project.
+Most of this homework is taken from [Tut04]
+-/
+
+import Mathlib.Data.Real.Basic
+import Mathlib.Tactic
+
+/-
+The next set of exercises uses divisibility in `ℤ` (beware the `∣` symbol which is
+not ASCII). This is done by writing `\|`.
+
+By definition, `a ∣ b := ∃ k, b = a * k`, so you can prove `a ∣ b` using the
+`use` tactic.
+-/
+-- Until the end of this file, `a`, `b` and `c` will denote integers, unless
+-- explicitly stated otherwise
+variable (a b c : ℤ)
+
+-- [TUT] 0029
+example (h₁ : a ∣ b) (h₂ : b ∣ c) : a ∣ c := by
+  sorry
+
+/-
+A very common pattern is to have an assumption or lemma asserting
+  `h : ∃ x, y = ...`
+and this is used through the combo:
+  `rcases h with ⟨x, hx⟩`
+  `rw hx at ...`
+The tactic `rcases` allows us to simplify the above combo when the
+name `hx` is replaced by the special name `rfl`, as in the following example.
+-/
+example (h1 : a ∣ b) (h2 : a ∣ c) : a ∣ b + c := by
+  rcases h1 with ⟨k, rfl⟩
+  rcases h2 with ⟨l, rfl⟩
+  use k + l
+  ring
+
+-- Now it's your turn! Prove the following:
+
+
+example : a ∣ b → a ∣ c → a ∣ b + c := by
+-- To get an even shorter proof, you can use the same `rfl` trick with
+-- the `rintro` tactic too. Try proving it both ways.
+  sorry
+
```

Comment:
```suggestion


-- We open the `Function` namespace to be able to use theorems and definitions about functions with shorter names, e.g. `Surjective f` instead of `Function.Surjective f`.
```